From 446d7675371e3cbf49b0be95ed0694ff137bab8f Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Mon, 24 Apr 2017 15:40:32 -0500
Subject: [PATCH 1727/5242] MXSCM-293-1 arm: dts: fixes for MX6 SCM devices on
 L4.9

commit  16a3cc5cd55423b3086e330a557ec70f0e137c6c from
https://source.codeaurora.org/external/imx/linux-imx.git

- replace legacy wakeup property with 'wakeup-source'
- add "jedec,spi-nor" flash compatible binding
- fix lcd display id
- fixig typo for gpios
- use external wdog_b reset instead of issuing a sw reset
- add off-on-delay for usdhc vmmc-supply regulator on mx6sxscm
- add missing sai1 and wdog1 configs for mx6sxscm
- fix vgen6 regulator maximum microvolt for mx6dqscm
- disable CAAM since cannot coexist with LCDIF for mx6sxscm
- add the reg into the rpmsg node for mx6sxscm

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts    |    4 ++--
 .../imx6dqscm-1gb-evb-interleave-android-ldo.dts   |    4 ++--
 arch/arm/boot/dts/imx6dqscm-qwks-rev2.dtsi         |    8 ++++----
 arch/arm/boot/dts/imx6sxscm-1gb-evb-lcdif1-ldo.dts |    6 +++++-
 arch/arm/boot/dts/imx6sxscm-1gb-evb-m4-ldo.dts     |    2 ++
 arch/arm/boot/dts/imx6sxscm-512mb-evb-m4-ldo.dts   |    3 ++-
 arch/arm/boot/dts/imx6sxscm-evb.dts                |   17 ++++++++++++-----
 7 files changed, 29 insertions(+), 15 deletions(-)

diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
index f0dcf7d..1013cf7 100644
--- a/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
@@ -32,7 +32,7 @@
 	internal_scm_flash: m25p80@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
-		compatible = "micron,n25q128a13";
+		compatible = "micron,n25q128a13", "jedec,spi-nor";
 		spi-max-frequency = <20000000>;
 		reg = <0>;
 	};
@@ -151,7 +151,7 @@
 
 			vgen6_reg: vgen6 {
 				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3150000>;
+				regulator-max-microvolt = <3300000>;
 				regulator-always-on;
 			};
 		};
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts
index e0ca328..5136cd7 100644
--- a/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts
@@ -28,7 +28,7 @@
 	internal_scm_flash: m25p80@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
-		compatible = "micron,n25q128a13";
+		compatible = "micron,n25q128a13", "jedec,spi-nor";
 		spi-max-frequency = <20000000>;
 		reg = <0>;
 	};
@@ -140,7 +140,7 @@
 
 			vgen6_reg: vgen6 {
 				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3150000>;
+				regulator-max-microvolt = <3300000>;
 				regulator-always-on;
 			};
 		};
diff --git a/arch/arm/boot/dts/imx6dqscm-qwks-rev2.dtsi b/arch/arm/boot/dts/imx6dqscm-qwks-rev2.dtsi
index e69302f..361fe62 100644
--- a/arch/arm/boot/dts/imx6dqscm-qwks-rev2.dtsi
+++ b/arch/arm/boot/dts/imx6dqscm-qwks-rev2.dtsi
@@ -125,7 +125,7 @@
 		power {
 			label = "Power Button";
 			gpios = <&gpio5 2 GPIO_ACTIVE_LOW>;
-			gpio-key,wakeup;
+			wakeup-source;
 			linux,code = <KEY_POWER>;
 		};
 	};
@@ -163,7 +163,7 @@
 	flash: m25p80@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
-		compatible = "micron,n25q128a13";
+		compatible = "micron,n25q128a13", "jedec,spi-nor";
 		spi-max-frequency = <20000000>;
 		reg = <0>;
 	};
@@ -286,7 +286,7 @@
 
 			vgen6_reg: vgen6 {
 				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3150000>;
+				regulator-max-microvolt = <3300000>;
 				regulator-always-on;
 			};
 		};
@@ -483,7 +483,7 @@
 
 &hdmi_core {
 	ipu_id = <0>;
-	disp_id = <0>;
+	disp_id = <1>;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6sxscm-1gb-evb-lcdif1-ldo.dts b/arch/arm/boot/dts/imx6sxscm-1gb-evb-lcdif1-ldo.dts
index 5b3e054..03ddbe9 100644
--- a/arch/arm/boot/dts/imx6sxscm-1gb-evb-lcdif1-ldo.dts
+++ b/arch/arm/boot/dts/imx6sxscm-1gb-evb-lcdif1-ldo.dts
@@ -35,4 +35,8 @@
 
 &ov5640 {
 	status = "disabled";
-};
\ No newline at end of file
+};
+
+&crypto {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6sxscm-1gb-evb-m4-ldo.dts b/arch/arm/boot/dts/imx6sxscm-1gb-evb-m4-ldo.dts
index 8839631..af0e8be 100644
--- a/arch/arm/boot/dts/imx6sxscm-1gb-evb-m4-ldo.dts
+++ b/arch/arm/boot/dts/imx6sxscm-1gb-evb-m4-ldo.dts
@@ -56,6 +56,8 @@
 };
 
 &rpmsg{
+	vdev-nums = <1>;
+	reg = <0xbfff0000 0x10000>;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6sxscm-512mb-evb-m4-ldo.dts b/arch/arm/boot/dts/imx6sxscm-512mb-evb-m4-ldo.dts
index a43f8d7..2408594 100644
--- a/arch/arm/boot/dts/imx6sxscm-512mb-evb-m4-ldo.dts
+++ b/arch/arm/boot/dts/imx6sxscm-512mb-evb-m4-ldo.dts
@@ -14,5 +14,6 @@
 };
 
 &rpmsg{
-	reg = <0x9FFF0000 0x8000>;
+	vdev-nums = <1>;
+	reg = <0x9FFF0000 0x10000>;
 };
diff --git a/arch/arm/boot/dts/imx6sxscm-evb.dts b/arch/arm/boot/dts/imx6sxscm-evb.dts
index a905b71..97651e7 100644
--- a/arch/arm/boot/dts/imx6sxscm-evb.dts
+++ b/arch/arm/boot/dts/imx6sxscm-evb.dts
@@ -85,6 +85,7 @@
 			regulator-min-microvolt = <3000000>;
 			regulator-max-microvolt = <3000000>;
 			gpio = <&gpio2 11 GPIO_ACTIVE_HIGH>;
+			off-on-delay = <20000>;
 			enable-active-high;
 		};
 
@@ -136,7 +137,7 @@
 			regulator-name = "peri_3v3";
 			regulator-min-microvolt = <3300000>;
 			regulator-max-microvolt = <3300000>;
-			gpios = <&gpio4 16 GPIO_ACTIVE_HIGH>;
+			gpio = <&gpio4 16 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 			regulator-always-on;
 		};
@@ -548,7 +549,7 @@
 	display = <&display0>;
 	status = "disabled";
 
-	display0: display {
+	display0: display@0 {
 		bits-per-pixel = <16>;
 		bus-width = <24>;
 
@@ -578,7 +579,7 @@
 	disp-dev = "ldb";
 	status = "okay";
 
-	display1: display {
+	display1: display@1 {
 		bits-per-pixel = <16>;
 		bus-width = <18>;
 	};
@@ -630,6 +631,12 @@
 	status = "okay";
 };
 
+&sai1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sai1>;
+	status = "disabled";
+};
+
 &spdif {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_spdif>;
@@ -668,7 +675,7 @@
 	wifi-host;
 	no-1-8-v;
 	keep-power-in-suspend;
-	enable-sdio-wakeup;
+	wakeup-source;
 	status = "okay";
 };
 
@@ -681,7 +688,7 @@
 	cd-gpios = <>;
 	wp-gpios = <>;
 	keep-power-in-suspend;
-	enable-sdio-wakeup;
+	wakeup-source;
 	pm-ignore-notify;
 	vmmc-supply = <>;
 	status = "okay";
-- 
1.7.9.5

